qiskit/algorithms/evolvers/classical_methods/scipy_imaginary_evolver.py

```diff
@@ -0,0 +1,143 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Classical Quantum Real Time Evolution."""
+from typing import Tuple, List, Optional
+import numpy as np
+import scipy.sparse as sp
+from qiskit import QuantumCircuit
+from qiskit.opflow import StateFn
+from qiskit.quantum_info.states import Statevector
+
+
+from ..evolution_problem import EvolutionProblem
+from ..evolution_result import EvolutionResult
+from ..imaginary_evolver import ImaginaryEvolver
+from .scipy_evolver import SciPyEvolver
+
+
+class SciPyImaginaryEvolver(ImaginaryEvolver, SciPyEvolver):
+    r"""Classical Evolver for imaginary time evolution.
+
+    Evolves an initial state :math:`|\Psi\rangle` for an imaginary time :math:`\tau = it`
+    under a Hamiltonian  :math:`H`, as provided in the ``evolution_problem``.
+
+    For each timestep we use a taylor expansion of :math:`\exp(- \Delta \tau H)` to evolve the system.
+
+    """
+
+    def __init__(
+        self,
+        timesteps: Optional[int] = 100,
```

Comment:
```suggestion
        timesteps: int = 100,
```
Optional means that None is also a valid value to use. I think here is just "optional" in the sense it has a default value so you do not need to explicitly provide it. That is not what Optional (with the capital O means) - the newer way of writing this would be `int | None`.

In the quantum evolvers I think this is called `num_timesteps` so we might want to be consistent. There None is supported because I believe given that value it will auto-compute a number of steps given the time from the problem.